### PR TITLE
Make configurable which steps report back to pull requests

### PR DESCRIPTION
**Pull request by AndreasBurger**

### CODE REVIEW

File: concourse/model/traits/pullrequest.py [hunk 1]
```diff
@@ -64,6 +66,12 @@ def replacement_label(self):
         doc='configures the policies to apply to pull-requests',
         type=PullRequestPolicies,
     ),
+    AttributeSpec.optional(
+        name='no-status-report',
```
Comment (ccwienk) [inline concourse/model/traits/pullrequest.py]: also: how about disable or suppress?

File: concourse/model/traits/pullrequest.py [hunk 2]
```diff
@@ -108,3 +119,15 @@ def process_pipeline_args(self, pipeline_args: JobVariant):
 
         # patch-in the updated repository
         pipeline_args._repos_dict[repo_name] = pr_repo
+
+        # patch the configured steps so that they do not report their status back to PRs
+        for step_name in self.trait.no_status_report():
+            print(step_name)
```
Comment (ccwienk) [inline concourse/model/traits/pullrequest.py]: rm debugging output

File: concourse/model/traits/pullrequest.py [hunk 3]
```diff
@@ -108,3 +119,15 @@ def process_pipeline_args(self, pipeline_args: JobVariant):
...
+            if not pipeline_args.has_step(step_name):
+                raise model.base.ModelValidationError(
+                    f"Reporting to pull requests was disabled for step '{step_name}', but no step "
+                    f"'{step_name}' was found in job '{pipeline_args.variant_name}'"
+                )
+            step = pipeline_args.step(step_name)
+            step._notification_policy = StepNotificationPolicy.NO_NOTIFICATION
+            print(step.notification_policy())
```
Comment (ccwienk) [inline concourse/model/traits/pullrequest.py]: ditto

**Comment by AndreasBurger** [comment]
Split this PR in two by moving out the refactoring commits.

**Note:** Merging this before PR #325 will break pipelines!

**Pull request merged**